### PR TITLE
Story/rhoaieng 2273 convert api list jobs

### PR DESCRIPTION
**Pull request by Gkrumbach07**
closes: https://issues.redhat.com/browse/RHOAIENG-2273

## Description

- migrates api to render pipeline run jobs table and delete modal

## How Has This Been Tested?

- tests pass
- create scheduled runs via the KF UI.
- test single delete
- test multi delete
- test that versions show up in row
- test enable/disable

## Test Impact

Cypress tests
Verify Empty State with no jobs
Verify Table shows rows
Verify all filters exist & write any related tests for those
Verify the enable / disable toggle in the rows
## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by Gkrumbach07** [comment]
WIP to write more tests via the description of the issue

**Comment by Gkrumbach07** [comment]
Added tests

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
Might need to get rid of the Experiment filter too

image

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
image
`"/apis/v2beta1/experiments/cc0d3cff-b612-4550-b393-a8e74dba5745"`

These are all requests for experiment -- the same experiment. Is there something up here that we need to look into? Seems like a bad fetch logic to do so many concurrent requests.

**Comment by andrewballantyne** [comment]
[We'll address the experiment fetching](https://github.com/opendatahub-io/odh-dashboard/pull/2482#pullrequestreview-1872640606) in another PR -- it is added network noise because of KFPv2 API needing to fetch Experiments. We had this problem with Jobs, we can address it the same way.

File: frontend/src/concepts/pipelines/context/usePipelineNamespaceCR.ts [hunk 1]
```diff
@@ -33,7 +33,14 @@ const usePipelineNamespaceCR = (namespace: string): FetchState<State> => {
   const callback = React.useCallback<FetchStateCallbackPromise<State>>(
     (opts) => {
       if (name) {
-        return getPipelinesCR(namespace, name, opts);
+        return getPipelinesCR(namespace, name, opts).catch((e) => {
+          if (e.statusObject?.code === 404) {
```
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/context/usePipelineNamespaceCR.ts:37]: Thanks for cleaning this up in this PR 👍

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**